smarty/sampler.py

```diff
@@ -407,23 +429,39 @@ def AddBetaSubstituentAtom(self, atom1type, bondset, atom2type):
         an exception if a beta substituent is attempted to be added to an atom1type which does not have an alpha 
         substituent.
         """
-        s = atom1type[0][atom1type[0].find("(")+1:atom1type[0].find(")")]
-        print s
+        #s = atom1type[0][atom1type[0].find("(")+1:atom1type[0].find(")")]
+        #if self.verbose: print s
+
+        # counting '[' tells us how many atoms are in the mix
+        count = atom1type[0].count('[')
```

Comment:
I've reworked beta substituent method. I was drawing out first and second beta substitutions when I realized the beta substitution always comes after the `]` on the alpha atom. 
I think this simplifies it a little bit, but uses the framework that already existed.